upcloud/resource_upcloud_objectstorage_test.go

```diff
@@ -349,16 +349,20 @@ func verifyObjectStorageExists(accessKey, secretKey, name string) resource.TestC
 
 func verifyObjectStorageDoesNotExist(accessKey, secretKey, name string) resource.TestCheckFunc {
 	return func(state *terraform.State) error {
-		time.Sleep(time.Second * 3)
-		exists, err := doesObjectStorageExists(state, accessKey, secretKey)
-		if err != nil {
-			if err.Error() == "could not find resources" {
-				return nil
+
+		for _, rs := range state.RootModule().Resources {
+			if rs.Type != "upcloud_storage" {
+				continue
+			}
+
+			client := testAccProvider.Meta().(*service.Service)
+			_, err := client.GetObjectStorageDetails(&request.GetObjectStorageDetailsRequest{
+				UUID: rs.Primary.ID,
+			})
+
+			if err == nil {
```

Comment:
Maybe we should also check that the error is what we actually expect (with [errors.Is](https://golang.org/pkg/errors/#Is) or just even err.Error() == 'expected string'), rather than just making sure it's not nil? 

Any other errors should also return some sort of an error to fail the test but it's a good practice to check the particular error that we expect this call to return if we depend on such behaviour.